src/app/shared/forms/new-vehicle-form/new-vehicle-form.component.ts

```diff
@@ -1,12 +1,80 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewEncapsulation, Output, EventEmitter } from '@angular/core';
+import { SelectItem, SelectItemGroup, ConfirmationService, Message, MessageService } from 'primeng/api';
+import { Vehicle } from '@app/shared/models/vehicle';
+import { Manufacturer } from '@app/shared/models/manufacturer';
+import { VehiclesService } from '@app/core/vehicles/vehicles.service';
+
 
 @Component({
   selector: 'app-new-vehicle-form',
   templateUrl: './new-vehicle-form.component.html',
-  styleUrls: ['./new-vehicle-form.component.css']
+  styleUrls: ['./new-vehicle-form.component.css'],
+  encapsulation: ViewEncapsulation.None,
 })
 export class NewVehicleFormComponent implements OnInit {
 
-ngOnInit(){
-}
+  @Output() public vehicleSubmitted: EventEmitter<Vehicle> = new EventEmitter<Vehicle>();
+  private vehicle: Vehicle = new Vehicle();
+  private currentYear: number = new Date().getFullYear();
+  private transmissions: SelectItem[];
+  private manufacturers: Manufacturer[] = [
+    new Manufacturer("Audi", ["A3", "A4", "A5", "A6"]),
+    new Manufacturer("BMW", ["M3", "M5"]),
+    new Manufacturer("Seat", ["Leon", "Toledo"]),
+  ];
+  private filteredManufacturers: any[];
+  private filteredModels: any[];
+
+  constructor(private vehiclesService: VehiclesService, private confirmationService: ConfirmationService, private messageService: MessageService) {
+    this.transmissions = [
```

Comment:
dlaczego to tu, a manufacturery przy polu?

Reply:
Wycofałem to z konstruktora :+1: